models/domains_tlds.php

```diff
@@ -896,13 +949,21 @@ private function assignConfigurableOptions($package_id, array $vars)
         );
 
         // Update configurable options
-        $options = ['dns_management', 'email_forwarding', 'id_protection', 'epp_code'];
+        foreach ($this->features as $option) {
 
-        foreach ($options as $option) {
             if (isset($company_settings['domains_' . $option . '_option_group'])) {
                 $option_group_id = $company_settings['domains_' . $option . '_option_group'];
 
                 if (isset($vars[$option]) && (bool)$vars[$option]) {
+                    if (!$registrar->supportsFeature($option)) {
```

Comment:
I noticed a typo on CORE-4345, the function was named supportsDnsManagment() instead of supportsDnsManagement(), causing it to always return an error when trying to enable DNS management, even if the module supports it.